src/vs/workbench/contrib/notebook/browser/view/cellPart.ts

```diff
@@ -57,49 +64,149 @@ export abstract class CellPart extends Disposable {
 	updateForExecutionState(element: ICellViewModel, e: ICellExecutionStateChangedEvent): void { }
 }
 
+export abstract class CellOverlayPart extends Disposable {
+	protected currentCell: ICellViewModel | undefined;
+	protected cellDisposables = new DisposableStore();
+
+	constructor() {
+		super();
+	}
+
+	/**
+	 * Prepare model for cell part rendering
+	 * No DOM operations recommended within this operation
+	 */
+	prepareRenderCell(element: ICellViewModel): void { }
+
+	/**
+	 * Update the DOM for the cell `element`
+	 */
+	renderCell(element: ICellViewModel): void {
+		this.currentCell = element;
+		this.didRenderCell(element);
+	}
+
+	protected didRenderCell(element: ICellViewModel): void { }
+
+	/**
+	 * Dispose any disposables generated from `didRenderCell`
+	 */
+	unrenderCell(element: ICellViewModel): void {
+		this.currentCell = undefined;
+		this.cellDisposables.clear();
+	}
+
+	/**
+	 * Update internal DOM (top positions) per cell layout info change
+	 * Note that a cell part doesn't need to call `DOM.scheduleNextFrame`,
+	 * the list view will ensure that layout call is invoked in the right frame
+	 */
+	updateInternalLayoutNow(element: ICellViewModel): void { }
+
+	/**
+	 * Update per cell state change
+	 */
+	updateState(element: ICellViewModel, e: CellViewModelStateChangeEvent): void { }
+
+	/**
+	 * Update per execution state change.
+	 */
+	updateForExecutionState(element: ICellViewModel, e: ICellExecutionStateChangedEvent): void { }
+}
+
 export class CellPartsCollection {
+	private _scheduledOverlayRendering: IDisposable | undefined;
```

Comment:
Do these need to be disposed of when the `CellPartsCollection` is disposed of?

Reply:
Nice catch, made CellPartsCollection disposable and can be disposed with the cell template.